lisa/analysis/pixel6.py

```diff
@@ -0,0 +1,100 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2015, ARM Limited and contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import pandas as pd
+import holoviews as hv
+from holoviews import opts
+
+from lisa.datautils import df_refit_index
+from lisa.analysis.base import TraceAnalysisBase
+from lisa.trace import requires_events
+from lisa.notebook import plot_signal
+
+P6_EMETER_CHAN_NAMES = {
+    'S4M_VDD_CPUCL0': 'little',
+    'S3M_VDD_CPUCL1': 'mid',
+    'S2M_VDD_CPUCL2': 'big',
+}
+
+class Pixel6Analysis(TraceAnalysisBase):
+    """
+    Support for Pixel 6-specific analysis data
+
+    :param trace: input Trace object
+    :type trace: :class:`trace.Trace`
+    """
+
+    name = 'pixel6'
+
+###############################################################################
+# DataFrame Getter Methods
+###############################################################################
+    @TraceAnalysisBase.cache
+    @requires_events('pixel6_emeter')
+    def df_power_meter(self):
+        """
+        Get the power meter readings across the trace.
+
+        :retuns: A :class:`pandas.DataFrame` with:
+            * A ``little`` column (readings from the little cluster)
+            * A ``mid`` column (readings from the mid cluster)
+            * A ``big`` column (readings from the big cluster)
+        """
+        df = self.trace.df_event('pixel6_emeter')
+        df = df.pivot_table(values='value', index='ts', columns='chan_name').reset_index().rename(columns=P6_EMETER_CHAN_NAMES)
+        channels = list(P6_EMETER_CHAN_NAMES.values())
+        df['ts_diff'] = df['ts'].diff()
+        # convert channels to power usage delta in mW
+        for channel in channels:
+            df[channel] = df[channel].diff() / df["ts_diff"]
+        # convert ts to seconds
+        df['Time'] = df['ts'] / 1000
+        df = df[['Time'] + channels].set_index('Time').iloc[1:]
```

Comment:
Why do we need `.iloc[1:]` ?

Reply:
Well what we're interested in is the delta between measurements which makes one of the rows just contain 'NaN's in the end so I was slicing it off. df_add_delta also shifts the deltas up so now the empty row is the last one and I replaced this with `.iloc[:-1]`.

Reply:
Using `df_add_delta(..., window=self.trace.window)` should fix that without having to discard anything (and it will work properly when slicing the trace instead of just resulting in a possibly empty dataframe for a slice where the signal is perfectly defined)

Reply:
Even with window=self.trace.window there were still NaN rows left so I'll just add a `.dropna()` at the end instead of the `iloc`.

Reply:
There likely is still an issue if you end up with NaN at the end. Maybe it's only because some deltas are `0` so we end up with a division by 0 but otherwise it's fishy. Can you provide the end of the dataframe right after groupby ?